plugins/urls/handlers/website.py

```diff
@@ -0,0 +1,266 @@
+
+            if user_agent:
+                headers["User-Agent"] = user_agent
+        else:
+            headers["User-Agent"] = ("Mozilla/5.0 (Windows NT 6.3; rv:36.0) "
+                                     "Gecko/20100101 Firefox/36.0")
+
+        domain_langs = context.get("config") \
+            .get("accept_language", {}) \
+            .get("domains", {})
+
+        if url.domain in domain_langs:
+            headers["Accept-Language"] = domain_langs.get(url.domain)
+        else:
+            headers["Accept-Language"] = context.get("config") \
+                .get("accept_language", {}) \
+                .get("default", "en")
+
+        session = self.get_session(url, context)
+        session.get(url.text, headers=headers) \
+            .addCallback(self.callback, url, context, session) \
+            .addErrback(self.errback, url, context, session)
+
+        return False
+
+    def teardown(self):
+        # Save all our cookie stores
+        if self.global_session is not None:
+            self.global_session.cookies.save(ignore_discard=True)
+            self.global_session.close()
+
+        for session in self.group_sessions.itervalues():
+            session.cookies.save(ignore_discard=True)
+            session.close()
+
+    def reload(self):
+        self.teardown()
+        self.group_sessions = {}
+
+        self.global_session = Session()
+        self.global_session.cookies = self.get_cookie_jar("/global.txt")
+        self.global_session.session_type = "global"
+        self.global_session.cookies.set_mode(
+            # self.plugin.config["sessions"]["cookies"]["global"]
+            self.plugin.config.get("sessions", {})
+                .get("cookies", {})
+                .get("global", "discard")
+        )
+
+    def callback(self, response, url, context, session):
+        self.plugin.logger.trace(
+            "Headers: {0}", list(response.headers)
+        )
+        self.plugin.logger.trace("HTTP code: {0}", response.status_code)
+
+        if "content-type" not in response.headers:
+            response.headers["Content-Type"] = ""
+
+        content_type = response.headers["content-type"].lower()
+
+        if ";" in content_type:
+            content_type = content_type.split(";")[0]
+
+        if content_type not in context["config"]["content_types"]:
+            self.plugin.logger.debug(
+                "Unsupported Content-Type: %s"
+                % response.headers["content-type"]
+            )
+            return  # Not a supported content-type
+
+        soup = BeautifulSoup(response.text)
+
+        if soup.title and soup.title.string:
+            title = soup.title.string.strip()
+            title = re.sub("\s+", " ", title)
+            title = title
+
+            context["event"].target.respond(
+                to_unicode('"{0}" at {1}'.format(
+                    to_bytes(title), to_bytes(url.domain)
+                ))
```

Comment:
Eh, either way, you should be converting bytes to unicode, not unicode to bytes then back to unicode.

Requests will convert the content to a unicode string based on the encoding given. If the site isn't doing that correctly, not a lot we can do about it. Looking at that link, the request header doesn't have an encoding listed, but the HTML does (shift_jis). It's pretty shitty to do that, but hey, Japan is like at least a decade behind when it comes to web development. We could use the raw response content (`content` rather than `text`) to pass into BS, but then that would mess up sites that do it decently and put the encoding in a header. Perhaps check if the response has an encoding (either header check, or the `encoding` attribute may be None, not sure), and if not, pass the raw content into BS instead. I'm not sure if BS actually attempts to extract an encoding from the HTML or if it just tried to guess, like requests does. It's sort of a chicken and egg problem when you need to parse something to find the encoding, but you need the encoding before you can parse it. Testing, will update...